operator/src/main/java/org/bf2/operator/managers/IngressControllerManager.java

```diff
@@ -432,6 +449,66 @@ void reconcileIngressControllers() {
+
+        ensureBlueprintRouteMatching(annotations, tslConfig, blueprintBaseName);
+    }
+
+    public void ensureBlueprintRouteMatching(Optional<Map<String, String>> annotations, Optional<TLSConfig> tlsConfig, String blueprintBaseName) {
+        // see findMatchingBlueprint https://github.com/openshift/router/blob/master/pkg/router/template/configmanager/haproxy/manager.go#L817
+        if (!Boolean.TRUE.equals(dynamicConfigManager) || blueprintRouteNamespace == null) {
+            return;
+        }
+
+        var orderedAnnotations = annotations.map(TreeMap::new);
+        MessageDigest stableIdDigest;
+        try {
+            stableIdDigest = MessageDigest.getInstance("SHA-1");
+            orderedAnnotations.ifPresent(m -> m.forEach((k, v) -> {
+                stableIdDigest.update(String.valueOf(k).getBytes());
+                stableIdDigest.update(String.valueOf(v).getBytes());
+            }));
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
+        }
+        tlsConfig.map(TLSConfig::getTermination).map(String::getBytes).ifPresent(stableIdDigest::update);
+        tlsConfig.map(TLSConfig::getCertificate).map(String::getBytes).ifPresent(stableIdDigest::update);
+        tlsConfig.map(TLSConfig::getKey).map(String::getBytes).ifPresent(stableIdDigest::update);
+        tlsConfig.map(TLSConfig::getCaCertificate).map(String::getBytes).ifPresent(stableIdDigest::update);
+
+        var stable = new Base32().encodeToString(stableIdDigest.digest()).toLowerCase().replaceFirst("=$", "");
+        var stableResourceName = String.format("%s-%s-blueprint", blueprintBaseName, stable);
+
+        var blueprintRouteLabels = OperandUtils.getDefaultLabels();
+        blueprintRouteLabels.put(OperandUtils.INGRESS_TYPE, OperandUtils.SHARDED);
+        blueprintRouteLabels.putAll(getRouteMatchLabels());
+
+        // strip InsecureEdgeTerminationPolicy from the placeholder, the operator doesn't consider it.
+       var config = new TLSConfigBuilder(tlsConfig.orElse(new TLSConfig())).withInsecureEdgeTerminationPolicy(null).build();
+        var blueprintRoute = new RouteBuilder()
+                .withNewMetadata()
+                .withName(stableResourceName)
+                .withLabels(blueprintRouteLabels)
+                .withAnnotations(orderedAnnotations.orElse(null))
+                .endMetadata()
+                .withNewSpec()
+                .withHost(stableResourceName)
+                .withNewPort()
+                .withTargetPort(new IntOrString("unused"))
+                .endPort()
+                .withTls(config)
+                .withNewTo()
+                .withKind("Service")
+                .withName("dummy")
+                .endTo()
+                .endSpec()
+                .build();
+
+        openShiftClient.routes().inNamespace(blueprintRouteNamespace).createOrReplace(blueprintRoute);
+    }
```

Comment:
I think that message is harmless.  @shawkins WDYT?

I think having the blueprint resource by a dependent on one kafka instance would lead to surprising behaviours.  However even if we thought the idea is good kube [disallows cross namespace ownership](https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/#:~:text=Cross%2Dnamespace%20owner%20references%20are,same%20namespace%20as%20the%20dependent.).

Reply:
@k-wall the message is harmless.  It will only show up when the operator first starts or when the blueprint is first created - I don't expect that those routes will be modified after that.  It could be avoided by not including OperandUtils.getDefaultLabels() in the blueprint labels - without the managed by label the informer won't consider it.

Reply:
It also looks like you've switched to using the fso namespace, in that case you can also consider setting the agent cr as the owner.

Reply:
@shawkins removing the defaultLabels seems to be least risk of surprise and therefore most appropriate for a defect fix release.